Observer/RecaptchaFormObserver.php

```diff
@@ -2,47 +2,53 @@
 
 namespace SY\Contact\Observer;
 
+use Magento\Framework\ObjectManagerInterface;
 use Magento\Framework\App\Response\Http;
 use Magento\Framework\App\Response\RedirectInterface;
 use Magento\Framework\Event\Observer;
 use Magento\Framework\Event\ObserverInterface;
-use Magento\ReCaptchaUi\Model\IsCaptchaEnabledInterface;
-use Magento\ReCaptchaUi\Model\RequestHandlerInterface;
 
 class RecaptchaFormObserver implements ObserverInterface
 {
     private const RECAPTCHA_FORM_KEY = 'sy_contact_form';
 
-    private IsCaptchaEnabledInterface $isCaptchaEnabled;
+    private $isCaptchaEnabled = null;
 
-    private RequestHandlerInterface $requestHandler;
+    private $requestHandler = null;
 
     private RedirectInterface $redirect;
 
     private Http $httpResponse;
 
     public function __construct(
-        IsCaptchaEnabledInterface $isCaptchaEnabled,
-        RequestHandlerInterface $requestHandler,
+        ObjectManagerInterface $objectManager,
         RedirectInterface $redirect,
         Http $httpResponse
     ) {
-        $this->isCaptchaEnabled = $isCaptchaEnabled;
-        $this->requestHandler = $requestHandler;
+        try {
+            $this->isCaptchaEnabled = $objectManager->get('Magento\ReCaptchaUi\Model\IsCaptchaEnabledInterface');
+            $this->requestHandler = $objectManager->get('Magento\ReCaptchaUi\Model\RequestHandlerInterface');
+        } catch (\ReflectionException $e) {
+            // ReCaptcha not available - we only weakly depend on it
+        }
         $this->redirect = $redirect;
         $this->httpResponse = $httpResponse;
     }
 
     public function execute(Observer $observer): void
     {
-        if ($this->isCaptchaEnabled->isCaptchaEnabledFor(self::RECAPTCHA_FORM_KEY)) {
-            $request = $observer->getRequest();
-            $this->requestHandler->execute(
-                self::RECAPTCHA_FORM_KEY,
-                $request,
-                $this->httpResponse,
-                $this->redirect->getRefererUrl()
-            );
+        if (!($this->isCaptchaEnabled !== null
+            && $this->requestHandler !== null
+            && $this->isCaptchaEnabled->isCaptchaEnabledFor(self::RECAPTCHA_FORM_KEY))) {
```

Comment:
Just formatting and with an `or` it seems a bit easier to read, but that is just nitpicking.
```
        if (
            $this->isCaptchaEnabled === null
            || $this->requestHandler === null
            || !$this->isCaptchaEnabled->isCaptchaEnabledFor(self::RECAPTCHA_FORM_KEY)
            ) {
```

Reply:
true, hm ..